Instationary navier stokes

sebglane opened the pull request:
A first very simple solver for the instationary navier stokes problem.

sebglane committed:
Small changes problem classes (CI probably fails)

Commit:
Continued with solver

sebglane committed:
Added rectangle mesh

sebglane committed:
Transient solver running

Commit:
Added test for hyper_rectangle and adjusted formatting according to PEP8 standard

sebglane committed:
Modified test: channel flow with time-dependent boundary condition

sebglane committed:
On instationary_navier_stokes: DFG benchmark mesh

sebglane committed:
index on instationary_navier_stokes: 0cb84dc Modified test: channel flow with time-dependent boundary condition

sebglane committed:
Merge branch 'instationary_navier_stokes' of github.com:idanven/Flow-problem-with-Fenics into instationary_navier_stokes

sebglane committed:
Merge commit 'stash@{0}' into instationary_navier_stokes

sebglane committed:
Added DFG benchmark without postprocessing

liuyiming0507 commented on source/grid_generator.py:
```diff
@@ -1,17 +1,12 @@
-import subprocess
-
+import dolfin as dlfn
 from enum import Enum, auto
```
I have a question here. 
Only this inheritance ```class (Enum)``` is defined, and then, ```BoundaryMarkers.left.value``` can be used in ```mark()```
```python
class BoundaryMarkers(Enum):    
      left = auto()
gamma01.mark(facet_marker, BoundaryMarkers.left.value)
```
if I don't define that class, just assign ```auto() ```to ```left``` directly, it can not work.
```python
left = auto()
gamma01.mark(facet_marker, left.value)
```
Why must we inherit ```Enum``` and then ```***.value```can work here

sebglane replied:
> Why must we inherit `Enum` and then `***.value`can work here

Well the purpose of the `auto` method is to automatically assign values to the members of an enumeration. The purpose of the enumeration here is provide an easy and understandable way of marking the boundaries for the user. It is not intended to be used outside of an `Enum`  class.

Quote from the python3 documentation: _An enumeration is a set of symbolic names (members) bound to unique, constant values. Within an enumeration, the members can be compared by identity, and the enumeration itself can be iterated over._

liuyiming0507 commented on source/navier_stokes_solver.py:
```diff
@@ -555,3 +579,347 @@ def solve(self):
+    _sub_space_association = {0: "velocity", 1: "pressure"}
+    _field_association = {value: key for key, value in _sub_space_association.items()}
+
+    def __init__(self, mesh, boundary_markers, time_stepping, tol, max_iter):
```
Why didn't set some default parameters here, like ```StationaryNavierStokesSolver```
```python
class StationaryNavierStokesSolver(NavierStokesSolverBase):   
      def __init__(self, mesh, boundary_markers, tol=1e-10, maxiter=50, tol_picard=1e-2, maxiter_picard=10):
```

sebglane replied:
In fact, we could use default values for the parameters `tol` and `max_iter`.

sebglane replied:
I added default values.

Commit:
PEP8 conforming formatting, added no cover statements

Commit:
Added Couette flow problem to stationary tests

Commit:
Added more tests for the stationary solver, fixed bug

Commit:
Excluded gmsh and meshio dependent parts from tests

Commit:
Update python-package-conda.yml

Removed duplicate run of the tests

Commit:
Update README.md

sebglane committed:
Removed unused code

Commit:
Fixed bug in test

sebglane commented:
I added more tests which increase the code coverage and added `# pragma: no cover` statements to exclude certain parts from the tests. Especially tests relying on `meshio` and `gmsh` are excluded.

Ready for review and possibly ready for merge.
